src/main/java/org/mandas/docker/client/DefaultDockerClient.java

```diff
@@ -2874,12 +2872,12 @@ public static Builder builder() {
    */
   public static Builder fromEnv() throws DockerCertificateException {
     final String endpoint = DockerHost.endpointFromEnv();
-    final Path dockerCertPath = Paths.get(Iterables.find(
-        Arrays.asList(DockerHost.certPathFromEnv(),
-            DockerHost.configPathFromEnv(),
-            DockerHost.defaultCertPath()),
-        Predicates.notNull()));
-
+    final Path dockerCertPath = Paths.get(asList(certPathFromEnv(), configPathFromEnv(), defaultCertPath())
+    	.stream()
+    	.filter(cert -> cert != null)
+    	.findFirst()
+    	.orElseThrow(() -> new NoSuchElementException("Cannot find docker certificated path")));
```

Comment:
I'm not very savvy with stream syntax, but this looks mostly reasonable.  The orElseThrow makes sense, but how does the Exception get handled when it occurs?

Reply:
The same way as before; the replaced `Iterables.find` throws a `NoSuchElementException` if the element is not found